src/dos/cdrom.h

```diff
@@ -410,8 +412,37 @@ class CDROM_Interface_Image final : public CDROM_Interface
 	static int           refCount;
 };
 
+class CDROM_Interface_Physical : public CDROM_Interface {
+public:
+	~CDROM_Interface_Physical() override;
+
+	bool GetAudioStatus(bool& playing, bool& pause) override;
+	bool PlayAudioSector(const uint32_t start, uint32_t len) override;
+	bool PauseAudio(bool resume) override;
+	bool StopAudio() override;
+	void ChannelControl(TCtrl ctrl) override;
+
+protected:
+	void InitAudio();
+
+private:
+	virtual std::vector<int16_t> ReadAudio(const uint32_t sector, const uint32_t frames_requested) = 0;
+	void CdAudioCallback(const uint16_t requested_frames);
+	void CdReaderLoop();
+
+	mixer_channel_t mixer_channel  = {};
+	std::thread thread             = {};
+	std::mutex mutex               = {};
+	std::condition_variable waiter = {};
+	RWQueue<AudioFrame> queue        {220500};
```

Comment:
Why this number?

Reply:
It's 5 seconds of audio.  Expanded to `REDBOOK_PCM_FRAMES_PER_SECOND * 5`